### PR TITLE
Update Laminas Coding Standard to 2.3, Remove Interop\Container, Drop Support for PHP 7.3

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Adds strict types
- Fixes calls to DomDocument#importNode to use `bool` for deep parameter
- Drops support for PHP 7.3
- Moves `laminas-servicemanager` into `require` at 3.14 minimum, making it a hard dependency and allowing the removal of `Interop\Container` and the ZF Bridge

### CODE REVIEW

File: src/Writer/Renderer/Entry/Rss.php [hunk 1]
```diff
@@ -218,7 +220,7 @@ protected function _setEnclosure(DOMDocument $dom, DOMElement $root)
         }
         $enclosure = $this->dom->createElement('enclosure');
         $enclosure->setAttribute('type', $data['type']);
-        $enclosure->setAttribute('length', $data['length']);
+        $enclosure->setAttribute('length', (string) $data['length']);
```
Comment (Ocramius) [inline src/Writer/Renderer/Entry/Rss.php:223]: Ok, this will likely not fly without a major release 😬

Comment (gsteel) [reply src/Writer/Renderer/Entry/Rss.php:223]: L201:L220 assert the value is an integer - why would this be a BC break?

**Comment by gsteel** [comment]
<del>This is going to need more work! Feel free to close it @Ocramius if you want to get 2.18 out the door</del>

File: src/Writer/Renderer/Entry/Rss.php (hunk 1 above)
Comment (Ocramius) [reply src/Writer/Renderer/Entry/Rss.php:223]: Ah, confused this with a test, sorry.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
:ship: from my end.

I will probably regret it, but I checked this multiple times, and we don't seem to be calling userland functions from our own implementation.

In case of crashes, I will apply fixes first, and if that also fails, I will revert the strict types in another patch.

**Comment by Ocramius** [comment]
Hah, it already [broke](https://github.com/laminas/laminas-feed/runs/7405125679?check_suite_focus=true) because `laminas/laminas-servicemanager` just got released (with `psr/container:^2` compat :D )

Awesome xD

**Comment by Ocramius** [comment]
Related: https://github.com/laminas/laminas-servicemanager/pull/127#issuecomment-1188754196

**Comment by Ocramius** [comment]
Fix provided in https://github.com/laminas/laminas-feed/pull/59

**Pull request merged**